internal/pkg/mqtt/sub.go

```diff
@@ -26,27 +43,58 @@ type broker struct {
 	svc    agent.Service
 	client paho.Client
 	logger logger.Logger
+	nats   *nats.Conn
 }
 
 // NewBroker returns new MQTT broker instance.
-func NewBroker(svc agent.Service, client paho.Client, log logger.Logger) MqttBroker {
+func NewBroker(svc agent.Service, client paho.Client, nats *nats.Conn, log logger.Logger) MqttBroker {
 	return &broker{
 		svc:    svc,
 		client: client,
 		logger: log,
+		nats:   nats,
 	}
 }
 
 // Subscribe subscribes to the MQTT message broker
 func (b *broker) Subscribe(topic string) error {
-	s := b.client.Subscribe(topic, 0, b.handleMsg)
+	s := b.client.Subscribe(fmt.Sprintf("%s/%s", topic, reqTopic), 0, b.handleMsg)
 	if err := s.Error(); s.Wait() && err != nil {
 		return err
 	}
 
+	if b.nats != nil {
+		n := b.client.Subscribe(fmt.Sprintf("%s/%s/#", topic, servTopic), 0, b.handleNatsMsg)
+		if err := n.Error(); n.Wait() && err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+// handleNatsMsg triggered when new message is received on MQTT broker
+func (b *broker) handleNatsMsg(mc paho.Client, msg paho.Message) {
+	topic := extractNatsTopic(msg.Topic())
+	if topic != "" {
+		b.nats.Publish(topic, msg.Payload())
+	}
+}
+
+func extractNatsTopic(topic string) string {
+	isEmpty := func(s string) bool {
+		return (len(s) == 0)
+	}
+	channelParts := channelPartRegExp.FindStringSubmatch(topic)
+	if len(channelParts) < 2 {
```

Comment:
This should be `len(channelParts) < 3` if `channelParts[2]`

Reply:
interesting catch 
i used this solution
https://github.com/mainflux/mainflux/blob/master/http/api/transport.go#L94

Reply:
I would fix it in both places